### PR TITLE
XLAB Add leaderboard API call

### PR DESCRIPTION
**Pull request by kcagran**
## Summary
 - AAP-88666
 - AAP-88665
 - AAP-88664
 - AAP-88663
 - AAP-88662
 - AAP-88667
This pull request introduces several improvements and fixes to the automation dashboard's leaderboard UI and its test coverage. The main changes include improved handling and display of ranking information, more robust test setups using mock data, and the addition of a new field to the collection status type to support future features.

**Leaderboard and Ranking Display Improvements:**

* The leaderboard UI now correctly hides the crown icon and rank label for dimensions or organizations where the user/org is unranked (rank 0), preventing misleading or empty rank information from being shown. (`AutomationDimensions.tsx`, `HighlightsLeaderboardPanel.tsx`) [[1]](diffhunk://#diff-9bce1e166bbae87e731441e98567c769a72df92e98c7ce1923be28f94852af3cL76-R88) [[2]](diffhunk://#diff-d049f2d4a00db59b37e92ac5556cca2d11247ba27d444c06858ddd9a4636750fR86-R94)

**Test Coverage and Mocking Enhancements:**

* Refactored leaderboard-related test files to use consistent, centralized mock data and reset state between tests, making tests more reliable and easier to maintain. (`AutomationDimensions.test.tsx`, `HighlightsLeaderboardPanel.test.tsx`, `MilestoneBadgesCard.test.tsx`) [[1]](diffhunk://#diff-141f9fed659a3983bf85a742cc5f1e47e592e548f80142fbb5a7ef62b725e22aL2-R48) [[2]](diffhunk://#diff-141f9fed659a3983bf85a742cc5f1e47e592e548f80142fbb5a7ef62b725e22aR78-R95) [[3]](diffhunk://#diff-023a494d6b565355da6d072cebc419e0c9141dab4ca635ce30ada46b774923e1L2-R38) [[4]](diffhunk://#diff-023a494d6b565355da6d072cebc419e0c9141dab4ca635ce30ada46b774923e1R49-R52) [[5]](diffhunk://#diff-6775190703b142565cab01481069b34dab278f4d9c40737ffe2f7558d4ca8ac9L2-R32)

* Added new test cases to verify that rank information is omitted from the UI when not applicable (i.e., rank 0). (`AutomationDimensions.test.tsx`, `HighlightsLeaderboardPanel.test.tsx`) [[1]](diffhunk://#diff-141f9fed659a3983bf85a742cc5f1e47e592e548f80142fbb5a7ef62b725e22aR78-R95) [[2]](diffhunk://#diff-023a494d6b565355da6d072cebc419e0c9141dab4ca635ce30ada46b774923e1R81-R89)

**Type and API Compatibility Updates:**

* Added an optional `min_collection_timestamp` field to the `IAutomationDashboardCollectionStatus` type and ensured it is passed through from the API and covered by tests. (`types/index.ts`, `useAutomationDashboardCollectionStatus.tsx`, `useAutomationDashboardCollectionStatus.test.ts`) [[1]](diffhunk://#diff-d05e212d7a2ec1b5d1cb032cd15b62af8941dc11a7c26eccd21941acfdf1c7dbR66) [[2]](diffhunk://#diff-4853801456815b2fe49367fd78474e5f2544a29324bc93f677df41dbdc6d9468R12) [[3]](diffhunk://#diff-e9942e0794401f2ef01e352fda86b38cce3153b6bb0d55bcb81f2c27d56a296dR23) [[4]](diffhunk://#diff-e9942e0794401f2ef01e352fda86b38cce3153b6bb0d55bcb81f2c27d56a296dR166-R182)

**Fixture and Mock Data Updates:**

* Updated the leaderboard fixture data to match the raw API shape, improving test realism and future maintainability. (`useAutomationLeaderboardsView.fixtures.ts`)

These changes collectively improve the reliability and clarity of leaderboard displays, strengthen test coverage, and lay groundwork for future enhancements.
<!-- What changed and why? -->

## Type of Change

- [ ] Bug fix
- [x] Enhancement
- [x] Tests
- [ ] Documentation
- [ ] Other (please specify)

## Risk Analysis - REQUIRED

<!-- SELECT ONE. This is required — the PR check will fail if none is selected. -->

- [ ] **High** — Broad platform impact (e.g., SWR config, shared framework components, authentication, routing, API wrappers, build/bundler config). Changes affect multiple workspaces or could cause widespread regressions.
- [x] **Medium** — Scoped but cross-cutting (e.g., shared utility functions, changes to multiple pages within one workspace, component library updates, test infrastructure). Limited blast radius but touches common code paths.
- [ ] **Low** — Narrowly scoped (e.g., single page fix, styling tweak, documentation, test-only changes, copy/string updates). Minimal risk of unintended side effects.

## Dependencies

<!-- List any dependent PRs, required backend changes, or manual setup steps. Remove this section if none. -->

## Testing

### Ephemeral E2E Tests

Once PR is ready and preliminary checks pass, trigger tests by posting a comment `/run-playwright` on this PR.

> Tests run against a fresh AAP instance (version based on branch).

### External Server E2E Runs

<!-- If applicable, attach run(s) from an external server using the GitHub Actions below. Mention the deployment type of the environment used.
- [Run Playwright with Currents](../actions/workflows/run-playwright-currents.yml)
-->

### Manual Testing

Open Automation Dashboard → Leaderboards (`analytics/automation-dashboard`).                                                                                                                                                                                                                                                                                 

**Sync status handling**
 - Verify no "never synced" empty state flashes while `collection_status` is still loading.
 - Make `/dashboard_reports/collection_status/` fail (e.g. block it in DevTools). Verify the page shows "Unable to load sync status" with the error message instead of the empty state.
 - Make `/dashboard_reports/leaderboard/` fail. Verify the leaderboard error state still shows (takes precedence). 
 
 **Null / missing API fields**
- Use an instance where no template ran in the last 30 days. Verify the page renders without crashing (featured template falls back to empty name / 0 runs).
- Log in as a user who does not belong to any organization. Verify the streak and org standing render without errors (org runs fall back to 0).
- Verify lists returned as `null` or omitted (leaderboards, streak daily, achievements, activity levels) render as empty, not as a crash.
 
**Automation Dimensions**
- User ranked in the top 10: verify the score and rank ("Rank X of Y") are shown, with a crown for ranks 1–3.
- User ranked but outside the top 10: verify the score shows "—" (not 0) and the rank is still shown.
- Unranked user (rank 0): verify the score shows 0 and no rank label or crown.

<!-- Steps to verify this change, if not obvious from the Jira issue. -->

### Screenshots
<img width="1321" height="861" alt="image" src="https://github.com/user-attachments/assets/7d0d4766-ad65-4680-9ff1-3f65bd8b9882" />

<img width="1321" height="861" alt="image" src="https://github.com/user-attachments/assets/6a18f541-d492-4c08-9dc9-907dcf5f1c76" />


<!-- If applicable. Any visual/UI changes MUST include before and after screenshots. -->

### CODE REVIEW

File: frontend/awx/analytics/automation-dashboard/views/useAutomationLeaderboardsView.ts [hunk 1]
```diff
@@ -100,8 +102,257 @@ export interface AutomationLeaderboardsView extends AutomationLeaderboardsData {
...
+}
+
 // ─── Hook ────────────────────────────────────────────────────────────────────
 
 export function useAutomationLeaderboardsView(): AutomationLeaderboardsView {
-  return { ...MOCK_LEADERBOARDS, isLoading: false, error: undefined };
+  const fetcher = useFetcher();
+
+  const leaderboardUrl = metricsAPI`/dashboard_reports/leaderboard/`;
+  const { data, error, isLoading } = useSWR<ILeaderboardReport, Error>(leaderboardUrl, fetcher);
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/views/useAutomationLeaderboardsView.ts:337]: **Medium — `isLoading` only tracks the leaderboard request**

`AutomationLeaderboards` gates the full UI on `lastSyncedAt`, but `isLoading` here comes only from the leaderboard SWR call. If the leaderboard resolves before `collection_status`, the tab can briefly show the "No leaderboard data yet" empty state (`isLoading: false`, `lastSyncedAt: null`) even though data is on the way.

This is more likely for system auditors: `useAutomationDashboardCollectionStatus` does not prefetch `collection_status` (platform-auditor gate), so both endpoints are fetched cold on the Leaderboards tab.

Consider combining loading states, e.g. `isLoading: isLeaderboardLoading || isCollectionStatusLoading`, or otherwise avoid treating a null `lastSyncedAt` as "never synced" while the status request is still in flight.

File: frontend/awx/analytics/automation-dashboard/views/useAutomationLeaderboardsView.ts [hunk 2]
```diff
@@ -100,8 +102,257 @@ export interface AutomationLeaderboardsView extends AutomationLeaderboardsData {
...
+  const collectionStatusUrl = metricsAPI`/dashboard_reports/collection_status/`;
+  const { data: collectionStatus } = useSWR<IAutomationDashboardCollectionStatus, Error>(
+    collectionStatusUrl,
+    fetcher
+  );
+
+  // Temporary workaround: collection_status doesn't expose a dedicated "last sync" timestamp
+  // yet, so min_collection_timestamp is used as a stand-in. Once the backend adds a proper
+  // last-sync timestamp field to collection_status, switch to reading that field instead.
+  const lastSyncedAt = toIsoString(collectionStatus?.min_collection_timestamp);
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/views/useAutomationLeaderboardsView.ts:351]: **Medium — `collection_status` errors are silently ignored**

Only `data` is read from the second SWR call; a failed `collection_status` leaves `lastSyncedAt` null with no surfaced error. Combined with `AutomationLeaderboards`'s `if (!lastSyncedAt) → empty state` gate, a successful leaderboard response can still show "No leaderboard data yet" when the real issue is the status endpoint.

Worth either surfacing a distinct error, or at least not conflating "never synced" with "couldn't fetch sync timestamp." A test for leaderboard-OK / status-failed would lock this in.

File: frontend/awx/analytics/automation-dashboard/types/index.ts [hunk 3]
```diff
@@ -63,6 +63,7 @@ export interface IAutomationDashboardCollectionStatus {
   enabled: boolean | null;
   next_run: Date | null;
   initial_collection_status: string | null;
+  min_collection_timestamp?: Date | null;
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/types/index.ts:66]: **Low — type doesn't match runtime JSON shape**

`useFetcher` returns raw `response.json()` with no date reviver, so `min_collection_timestamp` will arrive as an ISO **string** (same as the pre-existing `next_run` situation). The leaderboards hook defends via `toIsoString`, but typing this as `Date | null` is misleading for future consumers.

Consider `string | null` (or `string | Date | null` with a comment) and a test that uses the string shape the API actually returns.

File: frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardCollectionStatus.test.ts [hunk 4]
```diff
@@ -162,6 +163,23 @@ describe('useAutomationDashboardCollectionStatus', () => {
       expect(result.current.isLoading).toBe(false);
     });
 
+    test('should pass through min_collection_timestamp from the API response', () => {
+      const apiData: IAutomationDashboardCollectionStatus = {
+        enabled: true,
+        next_run: null,
+        initial_collection_status: 'completed',
+        min_collection_timestamp: new Date('2026-09-01T14:00:00Z'),
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardCollectionStatus.test.ts:171]: This test asserts a `Date` object, but the real fetcher won't parse JSON dates — the API value will be a string at runtime. Using `'2026-09-01T14:00:00.000Z'` here (and asserting pass-through of that string, or normalizing in the hook) would better match production behavior.

File: frontend/awx/analytics/automation-dashboard/components/leaderboards/AutomationDimensions.tsx [hunk 5]
```diff
@@ -73,15 +73,19 @@ function DimensionRow({
         </FlexItem>
         <FlexItem style={{ textAlign: 'right' }}>
           <MetricValue>
-            {rank <= 3 ? (
+            {rank && rank <= 3 ? (
```
Comment (lgalis) [inline frontend/awx/analytics/automation-dashboard/components/leaderboards/AutomationDimensions.tsx:76]: Nit: behavior is correct for rank `0`, but `rank > 0` reads more clearly than a truthy check and makes the "unranked" intent explicit. Same applies to the `{rank ? (` guard on line 84.

**Review by lgalis — COMMENTED** [review]
## Review summary

Solid work wiring the leaderboards tab to the real API — the `mapLeaderboardReport` layer and MSW-backed tests are well structured. See inline comments for actionable code feedback.

### Test gaps
- No test for **leaderboard resolved while `collection_status` is still pending** (empty-state flash).
- No test for **leaderboard OK + `collection_status` failure/null** (false "No leaderboard data yet").
- No test using a **string** `min_collection_timestamp` (actual JSON shape).

### Suggestions
- **Duplicated `baseView` mocks** across `AutomationDimensions.test.tsx`, `HighlightsLeaderboardPanel.test.tsx`, and `MilestoneBadgesCard.test.tsx` — consider a shared test helper to avoid contract drift.
- **PR template:** add before/after screenshots for rank-label/crown hiding, fill in manual testing steps, and trigger `/run-playwright` (label is still `e2e: pending`).

### What looks good
- Clean API → UI mapping; components stay decoupled from snake_case fields.
- Fixture mirrors raw API with deliberate edge cases (missing org day, user outside top 10, unknown badge IDs).
- Contract tests preserved in the new `.test.tsx` (with MSW).
- Rank-0 UX and Playwright mock updates are appropriate.

**Verdict:** Request changes — address the loading / empty-state interaction (inline) before merge.

**Comment by lgalis** [comment]
2 Questions - /cc @tiyiprh 

Medium — Ungated collection_status fetch may block non-auditor users
useAutomationDashboardCollectionStatus only fetches collection_status for superusers/auditors (isSuperuserOrAuditor ? url : null). The new leaderboards hook fetches the same endpoint for all users, ungated, to derive lastSyncedAt.

If the backend restricts collection_status to auditors (which the existing gating suggests), regular users who can open the Leaderboards tab would get collectionStatusError and see "Unable to load sync status" instead of leaderboard data — even when /leaderboard/ succeeds.

Recommendation: Confirm backend permissions with a non-auditor test account. If the endpoint is restricted, either:

expose a leaderboard-specific sync timestamp on /leaderboard/, or
degrade gracefully (show leaderboard data without the "Updated" line when collection_status returns 403).
Medium — Leaderboard data is fully hidden when only sync status fails
AutomationLeaderboards gates the entire tab on collectionStatusError before rendering content:

if (collectionStatusError) {
  return <CollectionStatusErrorState ... />;
}
When /leaderboard/ succeeds but /collection_status/ fails, users lose access to all leaderboard data. The hook still maps and holds the data (atAGlance.jobsRun === 1234 in tests), but the UI never shows it.

This may be intentional, but it is a significant UX tradeoff vs. showing data with a non-blocking sync warning. Worth confirming with product/backend.

**Comment by kcagran** [comment]
> 2 Questions - /cc @tiyiprh
> 
> Medium — Ungated collection_status fetch may block non-auditor users useAutomationDashboardCollectionStatus only fetches collection_status for superusers/auditors (isSuperuserOrAuditor ? url : null). The new leaderboards hook fetches the same endpoint for all users, ungated, to derive lastSyncedAt.
> 
> If the backend restricts collection_status to auditors (which the existing gating suggests), regular users who can open the Leaderboards tab would get collectionStatusError and see "Unable to load sync status" instead of leaderboard data — even when /leaderboard/ succeeds.
> 
> Recommendation: Confirm backend permissions with a non-auditor test account. If the endpoint is restricted, either:
> 
> expose a leaderboard-specific sync timestamp on /leaderboard/, or degrade gracefully (show leaderboard data without the "Updated" line when collection_status returns 403). Medium — Leaderboard data is fully hidden when only sync status fails AutomationLeaderboards gates the entire tab on collectionStatusError before rendering content:
> 
> if (collectionStatusError) { return <CollectionStatusErrorState ... />; } When /leaderboard/ succeeds but /collection_status/ fails, users lose access to all leaderboard data. The hook still maps and holds the data (atAGlance.jobsRun === 1234 in tests), but the UI never shows it.
> 
> This may be intentional, but it is a significant UX tradeoff vs. showing data with a non-blocking sync warning. Worth confirming with product/backend.

This will be implemented in the next PR, when we implement permissions.

**Review by lgalis — APPROVED** [review]
LGTM
